hoomd/md/pytest/test_rigid.py

```diff
@@ -453,3 +453,171 @@ def test_rigid_dof(
+    initial_snapshot = one_particle_snapshot_factory(particle_types=["A", "B"])
+    if initial_snapshot.communicator.rank == 0:
+        initial_snapshot.particles.moment_inertia[:] = [(1, 1, 1)]
+
+    sim = simulation_factory(initial_snapshot)
+    sim.seed = 5
+
+    rigid = md.constrain.Rigid()
+    rigid.body["A"] = valid_body_definition
+    rigid.create_bodies(sim.state)
+
+    constvol = md.methods.ConstantVolume(filter=hoomd.filter.Rigid())
+    constant_torque = hoomd.md.force.Constant(filter=hoomd.filter.All())
+    constant_torque.constant_torque["A"] = (0, 0, 1)
+    integrator = md.Integrator(
+        dt=0.005,
+        methods=[constvol],
+        forces=[constant_torque],
+        integrate_rotational_dof=True,
+    )
+    integrator.rigid = rigid
+    sim.operations += integrator
+    sim.run(1)
+    # compare the resulting tangential velocities with expected values
+    expected_velocity = [
+        [0, 0, 0],
+        [-6.25e-08, 5.00e-03, 0.00e00],
+        [6.25e-08, -5.00e-03, 0.00e00],
+        [5.00e-03, 6.25e-08, 0.00e00],
+        [-5.00e-03, -6.25e-08, 0.00e00],
+    ]
+    new_snapshot = sim.state.get_snapshot()
+    assert np.allclose(expected_velocity, new_snapshot.particles.velocity)
+
+
+def test_velocity_constituents_moving_body(
+    simulation_factory, one_particle_snapshot_factory, valid_body_definition
+):
+    """Test velocity of constituents updates for moving rigid body."""
+    initial_snapshot = one_particle_snapshot_factory(particle_types=["A", "B"])
+    if initial_snapshot.communicator.rank == 0:
+        initial_snapshot.particles.moment_inertia[:] = [(1, 1, 1)]
+        initial_snapshot.particles.velocity[:] = [(0, 0, 3)]
+
+    sim = simulation_factory(initial_snapshot)
+    sim.seed = 5
+
+    rigid = md.constrain.Rigid()
+    rigid.body["A"] = valid_body_definition
+    rigid.create_bodies(sim.state)
+
+    constvol = md.methods.ConstantVolume(filter=hoomd.filter.Rigid())
+    integrator = md.Integrator(
+        dt=0.005,
+        methods=[constvol],
+        integrate_rotational_dof=True,
+    )
+    integrator.rigid = rigid
+    sim.operations += integrator
+    sim.run(1)
+
+    new_snapshot = sim.state.get_snapshot()
+    expected_velocity = [[0, 0, 3]] * 5
+    assert np.allclose(expected_velocity, new_snapshot.particles.velocity)
+
+    constant_torque = hoomd.md.force.Constant(filter=hoomd.filter.All())
+    constant_torque.constant_torque["A"] = (0, 0, 1)
+    integrator.forces.append(constant_torque)
+
+    sim.run(1)
+    # compare the resulting tangential velocities with expected values
+    expected_velocity = [
+        [0, 0, 3],
+        [-6.25e-08, 5.00e-03, 3],
+        [6.25e-08, -5.00e-03, 3],
+        [5.00e-03, 6.25e-08, 3],
+        [-5.00e-03, -6.25e-08, 3],
+    ]
+    new_snapshot = sim.state.get_snapshot()
+    assert np.allclose(expected_velocity, new_snapshot.particles.velocity)
+
+
+@pytest.mark.parametrize(
```

Comment:
Should this test have a COM velocity as an extra parametrization?

Reply:
Added that option, and also fixed a bug that prevented it from using the correct moment of inertia for the dimer.